athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/examples/ExampleMetadataHandler.java

```diff
@@ -180,26 +184,70 @@ public ListSchemasResponse doListSchemaNames(BlockAllocator allocator, ListSchem
     }
 
     /**
-     * Returns a static list of TableNames. A connector for a real data source would likely query that source's metadata
-     * to create a real list of TableNames for the requested schema name.
+     * Returns a static list of TableNames and a nextToken String used for pagination. The nextToken can be null if
+     * there are no more tables to paginate, but otherwise its value should be obfuscated. A connector for a read data
+     * source would likely query that source's metadata to create a real list of TableNames for the requested schema name.
+     * The request (ListTablesRequest) includes a nextToken (String) and pageSize (int) attributes that can
+     * be used to paginate the response. The nextToken (request.getNextToken()) will be null if this is the first
+     * paginated request, or will contain a value of the starting point for the current request. pageSize (request
+     * .getPageSize()) will be -1 (ListTablesRequest.UNLIMITED_PAGE_SIZE_VALUE) for non-paginated requests (full
+     * list of tables should be returned), or it will be an integer value > 0. The latter indicates the maximum
+     * number of tables to return in the response (ListTablesResponse).
+     * The response (ListTablesResponse) also has a nextToken (String) attribute used for pagination. Its value
+     * should be set with the starting point for the next paginated request. If all tables have been processed, the
+     * value returned for nextToken should be null.
```

Comment:
its not helpful to copy paste the same large blocks of documentation everywhere. Please consider the context in which you are editing things as well as the existing documentation. Pagination is a fairly simple concept, this explaination it simply too large.  Perhaops the only useful minute detail would be to mentioned the UNLIMITED page size constant so implementers can ensure their impl handles such cases where a client wants it all at once.

Reply:
Reduced.